src/java/htsjdk/variant/variantcontext/filter/FilteringIterator.java

```diff
@@ -0,0 +1,119 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package htsjdk.variant.variantcontext.filter;
+
+import htsjdk.samtools.util.CloseableIterator;
+import htsjdk.samtools.util.CloserUtil;
+import htsjdk.variant.variantcontext.VariantContext;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * A simple filtering iterator for VariantContexts that takes a base iterator
+ * and a VariantContextFilter. Similar to SAMRecordFilter, but simpler since it doesn't need to worry about read-pairs.
+ */
+
+public class FilteringIterator implements CloseableIterator<VariantContext>, Iterable<VariantContext>{
+    private final Iterator<VariantContext> iterator;
+    private final VariantContextFilter filter;
+    private VariantContext next = null;
+
+    /**
+     * Constructor
+     *
+     * @param iterator the backing iterator
+     * @param filter   the filter (which may be a FilterAggregator)
+     */
+    public FilteringIterator(final Iterator<VariantContext> iterator, final VariantContextFilter filter) {
+        this.iterator = iterator;
+        this.filter = filter;
+        next = getNextRecord();
+    }
+
+    @Override
+    public void close() {
+        CloserUtil.close(iterator);
+    }
+
+    /**
+     * Returns true if the iteration has more elements.
+     *
+     * @return true if the iteration has more elements.  Otherwise returns false.
+     */
+    @Override
+    public boolean hasNext() {
+        return next != null;
+    }
+
+    /**
+     * Returns the next element in the iteration.
+     *
+     * @return the next element in the iteration
+     * @throws java.util.NoSuchElementException
+     *
+     */
+    @Override
+    public VariantContext next() {
+        if (next == null) {
+            throw new NoSuchElementException("Iterator has no more elements.");
+        }
+        final VariantContext result = next;
+        next = getNextRecord();
+        return result;
+    }
+
+    /**
+     * Required method for Iterator API.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public void remove() {
+        throw new UnsupportedOperationException("Remove() not supported by FilteringIterator");
+    }
+
+    /**
+     * Gets the next record from the underlying iterator that passes the filter
+     *
+     * @return SAMRecord the next filter-passing record
+     */
+    private VariantContext getNextRecord() {
```

Comment:
should this method be renamed? `getNextRecord` does not apply to `VariantContext`